### DISCUSSION TITLE
Finalizing ports for both Testnet and Mainnet

### DISCUSSION THREAD

**Discussion by amirvalhalla** [pactus-project/pactus#593] [General]
we need to finalize our ports for `GRPC`, `HTTP` etc... on `testnet` and `mainnet`.

@b00f 
@kehiy 
@Ja7ad

**Comment by kehiy** [comment]
I think all of them are already finalized!
what do you mean exactly about finalizeing?

**Reply by amirvalhalla** [reply]
e.g. default port of `Prometheus` is 9090 which was our grpc port before [metrics pull request](https://github.com/pactus-project/pactus/pull/588) was that so we changed the grpc port to 5105.
I think we should investigate more about our testnet & mainnet ports to prevent conflicting with the default port of other services.

**Reply by kehiy** [reply]
@b00f should we clear the last port for each module?
we can suggest our opinion (with reason) and then select one, finally, we update the code with a new port. (we can open a poll in [discussions](https://github.com/pactus-project/pactus/discussions/) )

or just set a meeting for it.

**Reply by b00f** [reply]
Can you provide a table of current ports (both Testnet and Mainnet) per modules and service?

**Reply by kehiy** [reply]
yes, I will add a comment here for current ports tonight.
also, I will add a suggested table.

**Comment by kehiy** [comment]
FINAL PORTS:

|  module | mainnet | testnet |
|----------|----------|----------|
|   GRPC |   `50051`   | `50052` |
|   GPRC Gateway  |   `8080`   |  `8080`  |
|    HTTP  |   `80`   |   `80`  |
|    Network  |   `21888`   | `21777`  |
|    nanomsg  |   `40899`   |  `40799`  |

**Reply by kehiy** [reply]
if it's okay, I can open an issue and PR to apply new ports.
also, we should add main net config function.

**Reply by amirvalhalla** [reply]
I can't entirely agree with `GRPC` on port `9090` because it's the default port of `Prometheus`. I think we still should use the `5105` port for `GRPC` Also we've added `Metrics` to our `HTTP` port which `Metrics Endpoint` by default is disabled and node owners can enable through our configuration. so I don't think we should use a different endpoint for the `Metrics`. 

@kehiy 
@b00f

**Reply by kehiy** [reply]
we can use `9090` for Prometheus and `5000` for GRPC
I will edit it

**Reply by amirvalhalla** [reply]
@kehiy 
I've edited my reply please review it.

**Reply by kehiy** [reply]
@amirvalhalla If the Metrics are just a part of HTTP and just a route of it, we can remove the Metric port in the table!
(or also change GRPC to 9090 again)

**Reply by amirvalhalla** [reply]
@kehiy 
yes, we should remote `Metrics` port table. about the GRPC  port, we can't use 9090.

**Reply by kehiy** [reply]
@amirvalhalla I update the table! I think it's good now.

@b00f what's your opinion?

**Reply by amirvalhalla** [reply]
@kehiy 
I still do not agree with `GRPC` port because `5000` port may be used by other services in the future. e.g. `UPNP` protocol by default on `5000`. I think `5105` port is good for `GRPC`

**Reply by kehiy** [reply]
@amirvalhalla we change it to `5105`
But, why we need UPNP in feature

**Reply by amirvalhalla** [reply]
@kehiy 
I knew, It just was an example, I don't think choosing a specific port like `5000` or `6000` etc... be a good option. because a lot of open-source services using these ports, may be in the future we have some challenges to change our ports, specifically in `mainnet`

**Reply by kehiy** [reply]
@amirvalhalla ok, I will change it and wait for @b00f to respond
If he agrees then I change ports

But I think grpc port should be easy memories

**Reply by b00f** [reply]
- Network for the Mainnet should set to 21888, It is same as PAC coin type in [SLIP-0044](https://github.com/satoshilabs/slips/blob/master/slip-0044.md)
- Port 9090 [unofficially](https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers) registered for Prometheus.  Better we don't use it.
- Default port for [swagger](https://swagger.io/docs/open-source-tools/swagger-ui/usage/installation/) is 8080. So let's keep 8080 for swagger and 80 for HTTP in Mainnet
- For Testnet Swagger port and HTTP should be user defined.
- We need to choose port for gRPC and NanoMsg for both Testnet and Mainnet

**Reply by kehiy** [reply]
@b00f I updated the list

**Comment by themantre** [comment]
@kehiy Let's finalize this discussion. Could you please add new table that has two columns (one for Mainnet and one for Testnet)
 For gRPC: Mainnet: 50051, Testnet 50052

**Reply by kehiy** [reply]
@themantre @b00f 
I added the final ports now!
(Additionally, we can change the nano message to: `9091`, if no, we have the final table).

**Reply by amirvalhalla** [reply]
@b00f  @kehiy 
everything looks good. :)

**Reply by b00f** [reply]
For GRPC, let's make it like: Mainnet: 50051, Testnet 50052
Both Testnet and Mainet have same ports for HTTP and GPRC Gateway. (Gateway is commonly use 8080, HTTP: 80)
Feel free to choose another port for nanomsg testnet. (not same as mainnet)

**Reply by kehiy** [reply]
@b00f 
have a look again
I think we can use this table now.

**Reply by b00f** [reply]
@kehiy With your permission I updated it.

**Reply by kehiy** [reply]
no problem, is that the final one?
can we have an issue for this and finalize them?

**Reply by b00f** [reply]
Please send a PR and update configs based on the new port numbers.
